examples/lock-app/silabs/src/AppTask.cpp

```diff
@@ -180,6 +184,10 @@ CHIP_ERROR AppTask::Init()
     sLockLED.Init(LOCK_STATE_LED);
     sLockLED.Set(state.Value() == DlLockState::kUnlocked);
 
+#ifdef DIC_ENABLE
+    chip::DICManager::Init();
```

Comment:
This definitively shouldn't go there. Base Application would be better IMO since it would apply to all examples at once.